lib/node_types/esp/src/device.h

```diff
@@ -120,51 +278,157 @@ class Subdevice {
+ * 
+ * 5. AUTOMATIC MQTT INTEGRATION
+ *    No manual MQTT code needed:
+ *    - Topics auto-generated from device and subdevice names
+ *    - Publishing happens automatically when values change
+ *    - Subscriptions set up automatically
+ *    - Message routing handled by DeviceManager
+ * 
+ * DEVICE LIFECYCLE
+ * ================
+ * 1. Construction: Device object created, registers with DeviceManager
+ * 2. start(): Initialize hardware (override this in derived classes)
+ * 3. measure(): Read sensor/update actuator (called at pollrate)
+ * 4. publish(): Send to MQTT (automatic when values change)
+ * 5. receive(): Handle commands (automatic routing from MQTT)
+ * 
+ * CREATING A NEW DEVICE TYPE
+ * ==========================
+ * To create a new device type:
+ * 1. Inherit from Device (or a specialized base like I2C_Device)
+ * 2. Call Device constructor with name and pollrate
+ * 3. Override start() to initialize hardware
+ * 4. Override measure() to read sensor or update actuator
+ * 5. Use value() to get/set subdevice values
+ * 6. Optionally add callbacks for filtering or change detection
+ */
 class Device {
     protected:
-        Fixed_Map<Subdevice, IOTEMPOWER_MAX_SUBDEVICES> subdevices;
+        Fixed_Map<Subdevice, IOTEMPOWER_MAX_SUBDEVICES> subdevices;  // List of subdevices
 #ifdef mqtt_discovery_prefix
+        // Home Assistant MQTT Discovery support
         // TODO: might have to move this to subdevice (if there are two values measured)
-        String discovery_config_topic;
-        String discovery_info;
+        String discovery_config_topic;  // Topic for discovery config
+        String discovery_info;          // JSON discovery payload
         void create_discovery_info(const String& type,
             bool state_topic=true, const char* state_on=NULL, const char* state_off=NULL,
             bool command_topic=false, const char* payload_on=NULL, const char* payload_off=NULL,
             const String& extra_json=String());
 #endif
     private:
-        bool requires_precision = false; // usually precision not needed (only for buffers)
-        Ustring name; // device name and mqtt-topic extension
-        bool _ignore_case = true;
-        bool _report_change = true;
-        bool _needs_publishing = false;
-        bool _retained = false; // allow setting retained for publishing
-
-
-        // This is a chain of callbacks called for every measuring iteration of the
-        // device.
-        // It gets passed the triggering device. 
-        // the last measured value will be in .value() or
-        // .value(n) with n denoting the subdevice.
-        // .measured_value can be modified.
-        // Returning false means that the value should not be considered.
-        // Returning true means that the (new) value in .measured_value is correct.
-        // TODO: commented example
+        // PRECISION CONTROL
+        // Some devices (like LED strips) need very precise timing
```

Comment:
the led strips are handled nicely by the NeoPixelBus library, however, we need sometimes precision for measuring inputs like the voltage curve for power meters - these values have to be quickly measured over a short interval to catch 1 or 2 full sinuses of the magnetic field of the measured AC current. So please use other example here.